Flaky test when running curl to a Flux URL

I ran into the following flaky test several times yesterday (network issues?).

Sample run:
https://github.com/weaveworks/weave-gitops/actions/runs/3762703203/jobs/6395638535

```
Run curl -s https://fluxcd.io/install.sh | sudo bash
  curl -s https://fluxcd.io/install.sh | sudo bash
  shell: /usr/bin/bash -e {0}
  env:
    CI_CONTAINER_REGISTRY: europe-west1-docker.pkg.dev
    CI_CONTAINER_REPOSITORY: europe-west1-docker.pkg.dev/weave-gitops-clusters/weave-gitops
    GOROOT: /opt/hostedtoolcache/go/1.19.4/x64
[INFO]  Downloading metadata https://api.github.com/repos/fluxcd/flux[2](https://github.com/weaveworks/weave-gitops/actions/runs/3762703203/jobs/6395638535#step:5:2)/releases/latest
Error: Process completed with exit code 22.
```

<img width="1491" alt="Screenshot 2022-12-23 at 22 12 38" src="https://user-images.githubusercontent.com/8824708/209405985-a6cb168b-7212-4997-b9f6-eb0f4aab6902.png">

After restarting the test, it succeeds.

--- a/.github/workflows/pr.yaml
+++ b/.github/workflows/pr.yaml
@@ -70,7 +70,10 @@ jobs:
         uses: actions/setup-go@v2
         with:
           go-version: ${{ matrix.go-version }}
-      - run: curl -s https://fluxcd.io/install.sh | sudo bash
+      - name: Setup Flux CLI
+        uses: fluxcd/flux2/action@main
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
       - run: make unit-tests
       # - run: make lib-test
 

